rust/src/lib.rs

```diff
@@ -1674,3 +1674,80 @@ fn get_index_version(full_version: &str, index: usize) -> Result<String, Error>
         .ok_or(anyhow!(format!("Wrong version: {}", full_version)))?
         .to_string())
 }
+
+// ----------------------------------------------------------
+// Exported functions
+// ----------------------------------------------------------
+
+// this is callback function to be called each time when rust wants to send log data
+type LogCallback = extern "C" fn(level: std::ffi::c_int, message: *const std::os::raw::c_char);
+
+#[repr(C)]
+pub struct WebDriverPathResult {
+    success: bool,
+    driver_path: *mut std::ffi::c_char,
+    error: *mut std::ffi::c_char,
+}
+
+// this is just an example how to expose function for external usage
+#[no_mangle]
+pub extern "C" fn get_dummy_webdriver_path(driver_name: *const std::ffi::c_char, log: LogCallback) -> WebDriverPathResult {
+    let result = std::panic::catch_unwind(|| {
+        
+        for i in 1..6 {
+            let message = std::ffi::CString::new("Hello, I am logging message").unwrap();
+            //let message = std::ffi::CString::new(String::from("A").repeat(10_000_000)).unwrap();
+            log(i, message.as_ptr());
+        }
+
+        //panic!("Intentional panic for testing");
```

Comment:
@bonigarcia can you help me here? We want to handle all panics in this block and propagate it in returning result. Seems I do it incorrectly.

Reply:
It has been moved to another file. I will request another review.